### DISCUSSION TITLE
Vercel serverless functions responding with 504s (No code changes or deployments)

### DISCUSSION THREAD

**Discussion by lusol** [vercel/community#3761] [Help] [answered]
### Summary

My team is suddenly experiencing Vercel serverless throwing 504s. We have never experienced this problem. When it started happening, we hadn't pushed any new code or changed any configs. We are completely stumped on this. If anyone has experienced anything similar before, please let us know how you fixed!

### Steps to Reproduce

Build does fail, but we can't access staging or load any pages. No idea where to start.
<img width="1007" alt="Screen Shot 2023-08-24 at 6 58 53 PM" src="https://github.com/vercel/community/assets/9649910/1ba342bc-c8e3-4b73-95ae-c2f0a804a389">

**Comment by robwatkiss** [comment]
We are facing exactly the same issue, for us it started with new deployments at around 1900 GMT yesterday. We've got our production site live again using an instant rollback but now can't release any updates. As yet unable to find a cause.

**Comment by robwatkiss** [comment]
@swarnava @amyegan Can either of you help here? I've also got a ticket open under 00153685

**Comment by swarnava** [comment]
Hi, can you share the URL where you are seeing 504 error?

**Reply by robwatkiss** [reply]
Last successful deployment: https://sg-storefront-7o4eru988-sungod.vercel.app/
First failing deployment: https://sg-storefront-hl10fs5hr-sungod.vercel.app/

**Reply by swarnava** [reply]
Thank you! Just to double check, redeploying working deployment is still causing the issue? Do you use `@vercel/node` runtime?

**Reply by robwatkiss** [reply]
That's correct, yes. The failing one above is a redeployment of the successful one.

We use both `@vercel/node` and `@nuxtjs/vercel-builder` (Nuxt 2)

**Reply by swarnava** [reply]
Hi,

Could you redeploy once more and let us know if you are still able to reproduce this?

**Reply by fliptaboada** [reply]
Hi, I'm still seeing the same issue, just tried a redeploy on my side, here's the url: https://the-angry-penguin.vercel.app/
This app was running for about a year without any commits / deploys, I triggered a redeploy yesterday just to test this out and the 504s started happening (build passes normally).
Here's the last working deployment (which is the same code as the failing one): https://the-angry-penguin-p23wpxydr-aestudio.vercel.app/
For context, the app is built with https://nullstack.app/

**Reply by robwatkiss** [reply]
Still the same issue having redeployed again @swarnava : https://sg-storefront-o8crx9ose-sungod.vercel.app/

**Reply by t3abdulg** [reply]
just curious, do you happen to be using express.js?

**Reply by edysegura** [reply]
We are facing the same issue above! Our previous working deployment isn't working in a redeploy and all our previews have this 504: GATEWAY_TIMEOUT

**Reply by swarnava** [reply]
Understood. We are still looking. Thank you for your patience.

**Reply by fliptaboada** [reply]
@t3abdulg yep, it is using the express deploy method

**Reply by swarnava** [reply]
@robwatkiss we made a change in your account. Can you redeploy and let us know it goes?

**Reply by flaviokicis** [reply]
@swarnava Any news regarding this issue? is there something we can do on our account? our team needs to push a production build as soon as possible

**Reply by ecklf** [reply]
Do you still experience the error when re-deploying? In case this isn't resolved please provide us your deployment id

**Reply by fliptaboada** [reply]
@ecklf seems to be working now, ty!
We were just unsure before if we should keep testing or not based on the last message from @swarnava.

**Reply by flaviokicis** [reply]
Oh, now it looks like it's back to normal! Thanks, guys!

**Comment by mmachatschek** [comment]
I'm cross posting here because we have a lot of errors since yesterday `2023-08-24 4PM CEST`, we had a production deployment three hours prior `2023-08-24 1PM CEST` which until then had no issue.

https://github.com/orgs/vercel/discussions/3764

Deployment ID 2wu8NwaVAH1sFgvfrfcmn4pqr1pZ

Our vercel logs didn't show any errors but we could find the culprit in our external logging tool axiom.

The error message is thrown inside an edge middleware in our case and returns a 503

```
[POST] [middleware: "src/middleware"] /api reason=INTERNAL_EDGE_FUNCTION_UNHANDLED_ERROR, status=500, upstream_status=503, user_error=false
```

To me this looks like an infrastructure issue

**Comment by jasondias9** [comment]
Hitting the same issue. Express server deploy method getting 504s on new deployments since yesterday around 15:00 EST. 
Old builds continue to work. 
Case number: #00153557

**Comment by julio-robles** [comment]
Same issue with express js.

**Comment by swarnava** [comment, marked as answer]
Alright folks. We reverted a change which we believe should fix the problem. Let us know if you are still facing trouble.

**Reply by lusol** [reply]
Thank you so much!!! working now

**Reply by edysegura** [reply]
It is working here as well!! Tks a lot!

**Reply by t3abdulg** [reply]
thank you for the fix!

**Comment by robwatkiss** [comment]
@swarnava I can confirm that reverting your change has resolved this issue.

I am extremely concerned, however, to see that the Vercel team have triggered builds on our production deployment without alerting us or confirming with us first. Especially when we've already raised issues with deployments failing - this had a very high likelihood or causing another outage, significant loss of revenue and ultimately, ruining my weekend.
